src/main/java/org/jahia/modules/sam/healthcheck/probes/JahiaExceptionsProbe.java

```diff
@@ -0,0 +1,81 @@
+package org.jahia.modules.sam.healthcheck.probes;
+
+import org.jahia.modules.sam.Probe;
+import org.jahia.modules.sam.ProbeSeverity;
+import org.jahia.modules.sam.ProbeStatus;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.text.MessageFormat;
+import java.util.Map;
+
+/**
+ * Probe to be used to get the number of exceptions in a log file.
+ * This probe have a LOW severity level.
+ * The probe returns a yellow level if there is at least one ERROR found in the log file.
+ * By default the log file which is used is located to System.getProperty("jahia.log.dir") + "jahia.log"
+ */
+@Component(immediate = true, service = Probe.class)
+public class JahiaExceptionsProbe implements Probe {
+
+    private static final Logger logger = LoggerFactory.getLogger(JahiaExceptionsProbe.class);
+
+    private String filePath;
+
+    private ProbeSeverity severity = ProbeSeverity.LOW;
+
+    private static final MessageFormat yellowMessage = new MessageFormat("A total of {0} exceptions are present on the platform, exceptions are not expected in a production environment and we recommend reviewing these.");
+
+    @Override
+    public String getName() {
+        return "JahiaExceptions";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Count the number of exceptions faced by Jahia";
+    }
+
+
+    @Override
+    public ProbeStatus getStatus() {
+        int numberOfError = 0;
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(filePath));
```

Comment:
Two questions there:
 - Are there performance implications if this file is very large ?
 - If the severity is set to Medium, can you confirm the probe's code is not executed ? (I just want to make sure we're not in a situation in which the probe's code is executed but only reported if the severity is set to low).

Reply:
In term of performance the BufferedReader seems to be a good option to read large files.

We doing the implement I tried two implementations the best one was the BufferedReader
java.util.Scanner took about 400ms to read my log file
java.io.BufferedReader took about 8ms to read my log file

I also found some resources which explains that using BufferedReader is a good option.

The probe is not called is the severity is set to MEDIUM, and the probe return something if it's set to LOW